Makefile

```diff
@@ -0,0 +1,87 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+COLLECTOR_CONTRIB_VERSION=0.99.0
+
+TOOLS = $(CURDIR)/.tools
+
+$(TOOLS):
+	mkdir -p $@
+
+YQ = $(TOOLS)/yq
+$(TOOLS)/yq: $(TOOLS)
+	curl -L https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 -o yq_linux_amd64 && \
+	chmod +x ./yq_linux_amd64 && \
+	mv ./yq_linux_amd64 $(TOOLS)/yq
+
+JQ = $(TOOLS)/jq
+$(TOOLS)/jq: $(TOOLS)
+	curl -L https://github.com/jqlang/jq/releases/latest/download/jq-linux-amd64 -o jq-linux-amd64 && \
+	chmod +x ./jq-linux-amd64 && \
+	mv ./jq-linux-amd64 $(TOOLS)/jq
+
+
+KUBECTL = $(TOOLS)/kubectl
+$(TOOLS)/kubectl: $(TOOLS)
+	curl -LO "https://dl.k8s.io/release/$(shell curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" && \
+	chmod +x ./kubectl && \
+	mv ./kubectl $(TOOLS)/
+
+GCLOUD_AUTH = $(TOOLS)/gcloud-auth
+$(TOOLS)/gcloud-auth: $(TOOLS)
+	#apt-get install -y google-cloud-sdk-gke-gcloud-auth-plugin
+
+.PHONY: tools
+tools: $(JQ) $(YQ) $(KUBECTL) $(GCLOUD_AUTH)
+
+.PHONY: generate
+generate: tools
+	$(KUBECTL) create configmap collector-config -n opentelemetry --from-file=./config/collector.yaml -o yaml --dry-run > ./k8s/base/1_configmap.yaml
+	$(YQ) -n 'load("config/collector.yaml") * load("test/collector.yaml")' > k8s/overlays/test/collector.yaml
+	cat test/fixtures/input.json | $(JQ) -c > k8s/overlays/test/fixture.json
+	$(KUBECTL) kustomize k8s/base > collector/collector.yaml
+
+.PHONY: generate-test
+generate-test: generate
```

Comment:
this was because I had the test manifests generated to the same place as the prod manifests (`collector/collector.yaml`). I'm removing that in favor of just using kustomize for test and prod, so this isn't needed anymore